### ISSUE TITLE
Disallow boolean typecasting for enums

### ISSUE THREAD

**Issue by Ky6uk** [palantir/tslint#4516]
### Rule Suggestion

Let's see a developer defined `enum` which will count from `0` by default:

```ts
const enum DogSays {
  WOOF, // 0
  BARK  // 1
}
```

After that the developer also defined a function:

```ts
function wasBorked(sound: DogSays | undefined) : boolean {
  if (!sound) {
    return false;
  }

  return true;
}
```

In this case the function will also return `false` when `sound` is `WOOF`. So I prefer to lint this place to use `if (sound === undefined) ...` instead of type casting. But I'm not sure it's possible without type checking. Maybe it's better to do it on the TS level instead of linter.

**Comment by JoshuaKGoldberg** [comment]
Hi @Ky6uk! This looks like it's covered by [`strict-boolean-expression`](https://palantir.github.io/tslint/rules/strict-boolean-expressions). Have you tried using that rule?

**Comment by Ky6uk** [comment]
@JoshuaKGoldberg Hmm, yes, that rule works but only in VSCode with latest `typescript-tslint-plugin`. And it doesn't work with jest. It seems because type info is required.

**Comment by JoshuaKGoldberg** [comment]
That's correct, type info is required. This rule suggestion would also require type information in order to determine the type of, for example, `sound` in the `if` statement.

Closing this as a subset of `strict-boolean-expressions`.
